library/lib.ml

```diff
@@ -328,16 +335,17 @@ let start_compilation s mp =
   lib_state := { !lib_state with comp_name = Some s;
                                  path_prefix = prefix }
 
+let open_blocks_message es =
+  let open_block_name = function
+      | oname, OpenedSection _ -> str "section " ++ pr_id (basename (fst oname))
+      | oname, OpenedModule (ty,_,_,_) -> str (module_kind ty) ++ spc () ++ pr_id (basename (fst oname))
+      | _ -> assert false in
+  str "The " ++ pr_enum open_block_name es ++ str " need to be closed."
```

Comment:
@tchajed: Yes, let's be realistic. Maybe, eventually, Coq will embed more and more natural language knowledge (starting e.g. with grammatically correct i18n), but let's not look too far in the future at the current time.

And thanks for the link, and, indirectly, for http://www.csse.monash.edu.au/~damian/papers/HTML/Plurals.html.